benchmarks/benchmarks/stats.py

```diff
@@ -74,15 +74,88 @@ def setup(self, nan_policy, method, variant):
         self.b = b
 
     def time_kendalltau(self, nan_policy, method, variant):
-        tau, p_value = stats.kendalltau(self.a, self.b, nan_policy=nan_policy, method=method, variant=variant)
+        tau, p_value = stats.kendalltau(self.a, self.b, nan_policy=nan_policy, 
+                                        method=method, variant=variant)
+
+
+class KS(Benchmark):
+    param_names = ['alternative', 'mode']
+    params = [
+        ['two-sided', 'less', 'greater'],
+        ['auto', 'exact', 'asymp'],
+    ]
+
+    def setup(self, alternative, mode):
+        rng = np.random.default_rng(0x2e7c964ff9a5cd6be22014c09f1dbba9)
+        self.a = stats.norm.rvs(loc=5, scale=10, size=500, random_state=rng)
+        self.b = stats.norm.rvs(loc=8, scale=10, size=500, random_state=rng)                                           
+    
+    def time_ks_1samp(self, alternative, mode):
+        stats.ks_1samp(self.a, stats.norm.cdf, 
+                       alternative=alternative, mode=mode)
+
+    def time_ks_2samp(self, alternative, mode):
+        stats.ks_2samp(self.a, self.b, alternative=alternative, mode=mode)
+
+
+class MannWhitneyU(Benchmark):
+    param_names = ['alternative', 'method']
+    params = [
+        ['two-sided', 'less', 'greater'],
+        ['auto', 'asymptotic', 'exact'],
+    ]
+
+    def setup(self, alternative, method):
+        rng = np.random.default_rng(0x2cedaefb4c66ab35f32d163376cc0c1b)
+        self.u1 = rng.uniform(-1, 1, 50)
+        self.u2 = rng.uniform(-0.5, 1.5, 100)
```

Comment:
This is indeed a bit slow. Only `exact` seems to depend on the array sizes here; the other two methods always take about 400 us. Would be good to check that that is the expected behavior.

Looking at the implementation, there's no immediately obvious place to improve performance (it was recently rewritten completely, and I think fairly carefully reviewed). May be useful to profile it though, to see where the time is spent.

Reply:
Being slower than other tests is not surprising. The exact method has a pure-Python, recursive calculation of the exact null distribution, and it was written primarily for clarity and to match the reference with only some attention to speed (i.e. memoizing results of method calls for huge performance gains, and memoizing manually rather than @memoize for further gains). The lowest hanging fruit would be to take advantage of the fact that the distribution is symmetric (see note [here](https://github.com/scipy/scipy/blob/ac3b21908adc937c42d99bd67b177723d484a101/scipy/stats/_mannwhitneyu.py#L38)). It might also benefit from being compiled, or perhaps there is a better formulation than the recursive one.

Reply:
> Note that the benchmark is not reporting accurate results because of the memoization. This test here actually takes about 3 seconds when run once on my machine.

This comment still needs addressing. If the function uses memoization, then I think we'd want the benchmark to just run for `1` repeat. And then add a comment explaining why.

Reply:
Also, using smaller input arrays so it takes in the 100 ms - 1 sec range would be nice.